### PR TITLE
Middleware that wraps any thrown exception in a CommandFailedException

### PR DESCRIPTION
**Pull request by frankverhoeven**
Adds a middleware that wraps any exception thrown during command execution in a `CommandFailedException`.

This allows to satisfy the `CommandBusInterface::handle` signature to only throw `CommandFailedException`s. It will also allow for improved decoupling of commands and handlers. Right now, various `commandbus::handle` calls catch exceptions that are specific for the given handler, this will no longer be needed.

### CODE REVIEW

File: tests/Infrastructure/League/Tactician/Middleware/CommandFailedWrapperTest.php [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+     * @var CommandFailedWrapper
+     */
+    private $middleware;
+
+    protected function setUp()
+    {
+        $this->middleware = new CommandFailedWrapper();
+    }
+
+    public function testWrapsAnyExceptionInCommandFailedException()
```
Comment (Feijs) [inline tests/Infrastructure/League/Tactician/Middleware/CommandFailedWrapperTest.php:31]: Perhaps you could also add a test that verifies the path where no exception is thrown

**Review by Feijs — COMMENTED** [review]
:+1:

File: tests/Infrastructure/League/Tactician/Middleware/CommandFailedWrapperTest.php (hunk 1 above)
Comment (frankverhoeven) [reply tests/Infrastructure/League/Tactician/Middleware/CommandFailedWrapperTest.php:31]: 👍🏼 Sure, will add Jenkins as well so scrutiniser can be removed

**Comment by frankverhoeven** [comment]
Could this get some love? ❤️
I propose to create a new major release for it, so it doesn't get into the monolith.

**Comment by frankverhoeven** [comment]
Closing in favour of symfony/messenger

**Pull request closed without merging**